felix/bpf/libbpf/libbpf_api.h

```diff
@@ -136,11 +136,13 @@ struct bpf_link *bpf_program_attach_cgroup(struct bpf_object *obj, int cgroup_fd
 
 	if (!(prog = bpf_object__find_program_by_name(obj, name))) {
 		err = ENOENT;
+		link = ERR_PTR(-ENOENT);
```

Comment:
Given that we're using errno, should we just set link to NULL if errno is non-zero?  That'd match Go's normal convention

Reply:
Yup, we got confused when I set link=NULL and then it kept failing, but the issue is that we do not rebuild when the cgo headers change. So fixing both.